deltacat/compute/stats/basic.py

```diff
@@ -1,94 +1,166 @@
 import ray
-from typing import Dict, Set, Tuple, List
+from typing import Dict, Set, Tuple, List, Optional
 
+from deltacat.compute.stats.models.dataset_stats import DatasetStats
 from ray.types import ObjectRef
 
 from deltacat.compute.stats.models.delta_stats_cache_result import DeltaStatsCacheResult
-from deltacat.compute.stats.models.stats_completion_info import StatsCompletionInfo
 from deltacat.compute.stats.models.stats_result import StatsResult
 from deltacat.compute.stats.types import StatsType
-from deltacat.compute.stats.utils.io import read_cached_delta_stats, cache_delta_stats, \
-    get_delta_stats
+from deltacat.compute.stats.utils.io import read_cached_delta_stats, cache_delta_column_stats, get_delta_stats, \
+    get_deltas_from_range
 from deltacat.compute.stats.utils.intervals import merge_intervals
 
-from deltacat.storage import PartitionLocator, DeltaLocator
+from deltacat.storage import PartitionLocator, DeltaLocator, Delta
 from deltacat.storage import interface as unimplemented_deltacat_storage
 
 
+# TODO (ricmiyam): Decouple DeltaCAT from S3-based paths
+# TODO (ricmiyam): Determine cache eviction policy
+
+
 def collect(
         source_partition_locator: PartitionLocator,
         delta_stream_position_range_set: Set[Tuple[int, int]],  # TODO: R&D on RangeSet impl
-        stat_types_to_collect: Set[StatsType],
-        stat_results_s3_bucket: str = None,  # TODO: Decouple DeltaCAT from S3-based paths
+        stat_types: Optional[Set[StatsType]],
+        columns: Optional[List[str]] = None,
+        stat_results_s3_bucket: Optional[str] = None,
         deltacat_storage=unimplemented_deltacat_storage) -> Dict[int, StatsResult]:
```

Comment:
In that case, it seems like `stat_types` is no longer necessary in this api since the current implementation of the stats collector will calculate and cache every stat regardless of this argument (this argument was only used to filter the aggregated stats at the end). 

Initially my idea was that fetching/storing by specific stats being requested can be de-prioritized since it would be a bit more effort to handle that. For now though it sounds like we're probably okay with just removing the `stat_types` here and letting the caller handle that themselves.

Reply:
I agree that we can get rid of stat types for now. Breaking things out by individual stat types requested will likely be helpful long-term, but we don't really need it for now.

We'll likely want to reintroduce it when we have statistic types to collect that are vastly more expensive to compute than others, and/or share no work overlap with other stat types.